File: benchmarks/bench_max_array.c
#include <stdio.h>
#include <stdint.h>

static int32_t maxarr(int32_t *a, int32_t n) {
    int32_t m = a[0];
    for (int32_t i = 1; i < n; i++) {
        if (a[i] > m) m = a[i];
    }
    return m;
}

int main(void) {
    int32_t arr[5] = {1, 5, 3, 2, 4};
    volatile int32_t r = 0;
    for (int i = 0; i < 1000000000; i++) {
        r = maxarr(arr, 5);
    }
    return (int)(r & 0xFF);
}

File: benchmarks/bench_max_array.zig
const std = @import("std");

fn maxarr(a: []const i32) i32 {
    var m: i32 = a[0];
    var i: usize = 1;
    while (i < a.len) : (i += 1) {
        if (a[i] > m) m = a[i];
    }
    return m;
}

pub fn main() u8 {
    const arr = [5]i32{ 1, 5, 3, 2, 4 };
    var r: i32 = 0;
    var i: i32 = 0;
    while (i < 1000000000) : (i += 1) {
        r = maxarr(&arr);
    }
    return @intCast(r & 0xFF);
}
